internal/locate/region_request.go

```diff
@@ -322,7 +322,19 @@ type accessKnownLeader struct {
 
 func (state *accessKnownLeader) next(bo *retry.Backoffer, selector *replicaSelector) (*RPCContext, error) {
 	leader := selector.replicas[state.leaderIdx]
-	if leader.isExhausted(maxReplicaAttempt) {
+	liveness := reachable
+	if atomic.LoadInt32(&leader.store.unreachable) != 0 {
+		liveness = selector.checkLiveness(bo, leader)
```

Comment:
Seems it's not necessary. `store.unreachable` means the liveness is `unreachable`.